aten/src/ATen/native/mkldnn/MKLDNNCommon.cpp

```diff
@@ -0,0 +1,66 @@
+#include <c10/core/OpaqueHandle.h>
+#include <c10/core/Allocator.h>
+#include "MKLDNNCommon.h"
+
+#if AT_MKLDNN_ENABLED()
+
+#include <ideep.hpp>
+
+namespace at { namespace native {
+
+// Custom allocator using c10 CPU allocator for `ideep::tensor`
+struct AllocForMKLDNN {
+  template<class computation_t = void>
+  static char* malloc(size_t size) {
+    auto allocator = c10::GetAllocator(c10::DeviceType::CPU);
+    return (char*)allocator->raw_allocate(size);
+  }
+
+  template<class computation_t = void>
+  static void free(void* p) {
+    auto allocator = c10::GetAllocator(c10::DeviceType::CPU);
+    allocator->raw_deallocate(p);
+  }
+};
+
+// Helper function to construct a `Storage` given allocated `ideep::tensor`.
+// The storage does not own the buffer. The assumption is that there would be
+// no reallocation from `ideep::tensor` later.
+c10::Storage new_with_itensor_storage(const ideep::tensor& it, const TensorOptions& options) {
+  c10::DataPtr data_ptr(it.get_data_handle(), c10::DeviceType::CPU);
```

Comment:
why do we even initialize storage pointer? from what I understand, nothing should be calling .data() on tensor (though you'd still need to initialize storage because of how accessors are organized today)

Reply:
@dzhulgakov The reason why I made storage initialized is that there is code making use of the `data_ptr`. For example:

- Some PyTorch code uses it to check Tensor equality, e.g. https://github.com/pytorch/fairseq/blob/master/fairseq/modules/multihead_attention.py#L75
- C10D uses it to access Tensor buffer for communication, e.g. https://github.com/pytorch/pytorch/blob/master/torch/lib/c10d/ProcessGroupGloo.cpp#L1391